### PR TITLE
Remove nodeSelector support for egress gateway for `az aks mesh` command.

### PR DESCRIPTION
**Pull request by deveshdama**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### Related command
`az aks mesh enable-egress-gateway -g myRG -n myCluster`

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally? (`pip install wheel==0.30.0` required)
- [x] My extension version conforms to the [Extension version schema](https://github.com/Azure/azure-cli/blob/release/doc/extensions/versioning_guidelines.md)

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

File: src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py [hunk 1]
```diff
@@ -11769,41 +11769,12 @@ def test_aks_azure_service_mesh_with_egress_gateway(
...
         try:
             self.cmd(
                 update_cmd,
                 checks=[
                     self.check("serviceMeshProfile.mode", "Istio"),
-                    self.check(
-                        "serviceMeshProfile.istio.components.egressGateways[0].nodeSelector.istio",
-                        "egress",
-                    ),
-                    self.check(
```
Comment (nshankar13) [inline src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py:11783]: We don't need to remove the basic enabled check right?

File: src/aks-preview/azext_aks_preview/tests/latest/test_managed_cluster_decorator.py [hunk 2]
```diff
@@ -7152,8 +7151,7 @@ def test_update_service_mesh_profile(self):
             self.client,
             {
                 "enable_azure_service_mesh": True,
-                "enable_egress_gateway": True,
-                "egress_gateway_nodeselector": "istio=egress",
+                "enable_egress_gateway": True
```
Comment (nshankar13) [inline src/aks-preview/azext_aks_preview/tests/latest/test_managed_cluster_decorator.py:7154]: Think we need a comma at the end of this line

File: src/aks-preview/azext_aks_preview/custom.py [hunk 3]
```diff
@@ -2913,16 +2913,14 @@ def aks_mesh_enable_egress_gateway(
         cmd,
         client,
         resource_group_name,
-        name,
-        egx_gtw_nodeselector,
+        name
```
Comment (nshankar13) [inline src/aks-preview/azext_aks_preview/custom.py:2916]: Comma here as well?

File: src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py (hunk 1 above)
Comment (deveshdama) [reply src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py:11783]: As discussed offline, this does the basic enabled check on line 11778

File: src/aks-preview/HISTORY.rst [hunk 4]
```diff
@@ -12,6 +12,10 @@ To release a new version, please select a new version number (usually plus 1 to
 Pending
 +++++++
 
+2.0.0b8
```
Comment (FumingZhang) [inline src/aks-preview/HISTORY.rst:15]: This should be a breaking change. According to [Extension version schema](https://github.com/Azure/azure-cli/blob/release/doc/extensions/versioning_guidelines.md), please choose `3.0.0b1`.

File: src/aks-preview/HISTORY.rst [hunk 5]
```diff
@@ -12,6 +12,10 @@ To release a new version, please select a new version number (usually plus 1 to
...
++++++++
+* Remove support for nodeSelector for egress gateway for `az aks mesh` command.
```
Comment (FumingZhang) [inline src/aks-preview/HISTORY.rst:17]: Make the note start with `[BREAKING CHANGE]`

File: src/aks-preview/setup.py [hunk 6]
```diff
@@ -9,7 +9,7 @@
 
 from setuptools import setup, find_packages
 
-VERSION = "2.0.0b7"
+VERSION = "2.0.0b8"
```
Comment (FumingZhang) [inline src/aks-preview/setup.py:12]: Update to `3.0.0b1`

**Review by FumingZhang — COMMENTED** [review]
Queued [live test](https://dev.azure.com/msazure/CloudNativeCompute/_build/results?buildId=90585353&view=results) to validate the change.
- test_aks_azure_service_mesh_enable_disable
- test_aks_azure_service_mesh_with_ingress_gateway
- test_aks_azure_service_mesh_with_egress_gateway
- test_aks_azure_service_mesh_with_pluginca
- test_aks_azure_service_mesh_get_revisions
- test_aks_azure_service_mesh_get_upgrades

BTW, case `test_aks_azure_service_mesh_get_revisions` is still filtered out by default (src/aks-preview/azcli_aks_live_test/configs/ext_matrix_default.json), do you think it's the time to add it back?
```
        "service mesh, missing feature registration for new resource type": [
            "test_aks_azure_service_mesh_get_revisions"
        ],
```

**Comment by deveshdama** [comment]
> Queued [live test](https://dev.azure.com/msazure/CloudNativeCompute/_build/results?buildId=90585353&view=results) to validate the change.
> 
> * test_aks_azure_service_mesh_enable_disable
> * test_aks_azure_service_mesh_with_ingress_gateway
> * test_aks_azure_service_mesh_with_egress_gateway
> * test_aks_azure_service_mesh_with_pluginca
> * test_aks_azure_service_mesh_get_revisions
> * test_aks_azure_service_mesh_get_upgrades
> 
> BTW, case `test_aks_azure_service_mesh_get_revisions` is still filtered out by default (src/aks-preview/azcli_aks_live_test/configs/ext_matrix_default.json), do you think it's the time to add it back?
> 
> ```
>         "service mesh, missing feature registration for new resource type": [
>             "test_aks_azure_service_mesh_get_revisions"
>         ],
> ```

File: src/aks-preview/azext_aks_preview/_help.py [hunk 7]
```diff
@@ -2911,12 +2911,9 @@
     short-summary: Enable an Azure Service Mesh egress gateway.
     long-summary: This command enables an Azure Service Mesh egress gateway in given cluster.
     parameters:
```
Comment (FumingZhang) [inline src/aks-preview/azext_aks_preview/_help.py:2913]: remove this line
```suggestion
```

**Review by nshankar13 — APPROVED** [review]

**Review by FumingZhang — APPROVED** [review]

**Review by yanzhudd — APPROVED** [review]

**Pull request merged**